src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -155,32 +155,52 @@ public void shootAtPlayer() {
         playerShot(coor);
     }
 
-    void playerShot(Coordinate coor) {
+    int playerShot(Coordinate coor) {
         if(playerMisses.contains(coor)){
             System.out.println("Dupe");
         }
 
         if(aircraftCarrier.covers(coor)){
             playerHits.add(coor);
-            checkSunk(aircraftCarrier, playerHits, playerSunk);
-        }else if (battleship.covers(coor)){
+            if(checkSunk(aircraftCarrier, playerHits, playerSunk))
```

Comment:
This checkSunk function should have an accompanying checkWin function, so that if all ships are sunk on one side of the game it should be reported to the player if they have won or lost.